pkg/consolegraphql/agent/agent.go

```diff
@@ -0,0 +1,534 @@
+					body := msg.Value.(map[string]interface{})
+					agentcon, err := FromAgentConnectionBody(body)
+					if err != nil {
+						log.Printf("ignoring failure unmarsall connection message %s for infraUuid %s, %v", body, aad.infraUuid, err)
+						break
+					} else {
+						evt := AgentEvent{
+							AddressSpaceNamespace: aad.addressSpaceNamespace,
+							AddressSpace:          aad.addressSpace,
+							InfraUuid:             aad.infraUuid,
+							Type:                  AgentEventInsertOrUpdateType,
+							Object:                agentcon,
+						}
+						err = aad.handler(evt)
+						if err != nil {
+							return err
+						}
+					}
+				case "connection_deleted":
+					body := msg.Value.(map[string]interface{})
+					agentcon, err := FromAgentConnectionBody(body)
+					if err != nil {
+						log.Printf("ignoring failure unmarsall connection message %s for infraUuid %s %v", body, aad.infraUuid, err)
+						break
+					} else {
+						evt := AgentEvent{
+							AddressSpaceNamespace: aad.addressSpaceNamespace,
+							AddressSpace:          aad.addressSpace,
+							InfraUuid:             aad.infraUuid,
+							Type:                  AgentEventTypeDelete,
+							Object:                agentcon,
+						}
+						err = aad.handler(evt)
+						if err != nil {
+							return err
+						}
+					}
+				case "address":
+					body := msg.Value.(map[string]interface{})
+					agentaddr, err := FromAgentAddressBody(body)
+					if err != nil {
+						log.Printf("ignoring failure unmarsall address message %s for infraUuid %s, %v", body, aad.infraUuid, err)
+						break
+					} else {
+						evt := AgentEvent{
+							AddressSpaceNamespace: aad.addressSpaceNamespace,
+							AddressSpace:          aad.addressSpace,
+							InfraUuid:             aad.infraUuid,
+							Type:                  AgentEventInsertOrUpdateType,
+							Object:                agentaddr,
+						}
+						err = aad.handler(evt)
+						if err != nil {
+							return err
+						}
+
+					}
+
+				case "address_deleted":
+				default:
+					// Ignore messages with other subjects
+				}
+
+				// Accept message
+				err = msg.Accept()
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+}
+
+func (aad *amqpAgentDelegate) CommandDelegate(bearerToken string) (CommandDelegate, error) {
+	aad.delegateMux.Lock()
+	defer aad.delegateMux.Unlock()
+
+	key := getShaSum(bearerToken)
+
+	now := time.Now()
+	if pair, present := aad.commandDelegates[key]; !present {
+		delegate := aad.newAgentDelegate(bearerToken)
+		aad.commandDelegates[key] = commandDelegatePair{
+			delegate: delegate,
+			lastUsed: now,
+		}
+		return delegate, nil
+	} else {
+		pair.lastUsed = now
+		return pair.delegate, nil
+	}
+}
+
+func (aad *amqpAgentDelegate) housekeepCommandDelegates() {
+	findExpiredCommandDelegates := func() []CommandDelegate {
+		aad.delegateMux.Lock()
+		defer aad.delegateMux.Unlock()
+
+		expired := make([]CommandDelegate, 0)
+
+		horizon := time.Now().Add(time.Minute * -5)
```

Comment:
Is this how long the collector outstanding commants are kept?

Reply:
Resolved above.